sys/aarch64/switch.S

```diff
@@ -47,6 +75,32 @@ ENTRY(ctx_switch)
 
         # save context of @from thread
 .ctx_save:
+        ldr     w3, [x0, #TD_PFLAGS]
+        and     w2, w3, #TDP_FPUINUSE|TDP_FPUCTXSAVED
+        mov     w4, #TDP_FPUINUSE
+        cmp     w2, w4
+        bne     .skip_fpu_save
+
+        orr     w3, w3, #TDP_FPUCTXSAVED
+        str     w3, [x0, #TD_PFLAGS]
+
+        /* enable FPU and save context */
+        /* thread_t::tdp_flags is a volatile unsigned - use 32-bit */
+        mrs     x2, cpacr_el1
+        and     x2, x2, ~CPACR_FPEN_MASK
+        orr     x2, x2, CPACR_FPEN_TRAP_NONE
+        msr     cpacr_el1, x2
+
+        ldr     x2, [x0, #TD_UCTX]
+        save_fpu_ctx x2, x3, w3
```

Comment:
Here you should also add a comment about 2nd and 3rd argument as you did before.